chainstate/test-framework/src/framework.rs

```diff
@@ -19,15 +19,21 @@ use chainstate_types::{BlockIndex, GenBlockIndex};
 use common::{
     chain::{Block, GenBlock, Genesis},
     primitives::{id::WithId, BlockHeight, Id, Idable},
+    time_getter::TimeGetter,
 };
 use crypto::random::Rng;
-use std::sync::Arc;
+use std::{
+    sync::{atomic::AtomicU64, Arc},
+    time::Duration,
+};
 
 /// The `Chainstate` wrapper that simplifies operations and checks in the tests.
 pub struct TestFramework {
     pub chainstate: super::TestChainstate,
     pub storage: TestStore,
     pub block_indexes: Vec<BlockIndex>,
+    pub time_getter: TimeGetter, // A clone of the TimeGetter supplied to the chainstate
+    pub time_value: Option<Arc<AtomicU64>>, // current time since epoch; if None, it means a custom TimeGetter was supplied and this is useless
```

Comment:
Minor: I'd prefer the comments to be before the struct member. This makes the lines a bit too long